test/unit/conditions/base/contract.test.ts

```diff
@@ -7,7 +7,7 @@ import {
 import { ContractCondition } from '../../../../src/conditions/base';
```

Comment:
I'd love to see some comprehensive testing around validation of various different functionAbis, both valid and invalid, as part of this PR.